src/components/AddItem.jsx

```diff
@@ -24,37 +25,73 @@ function AddItem() {
   const [itemName, setItemName] = useState('');
   const [frequency, setFrequency] = useState(7);
   const [notification, setNotification] = useState('');
+  const [list, setList] = useState([]);
+  const localToken = localStorage.getItem('list-token');
+  const [duplicateMessage, setDuplicateMessage] = useState(null);
 
-  //retrive the token from localStorage
-  const token = localStorage.getItem('list-token');
+  //useEffect to setList of items in that user's list which will be used for duplicate comparison
+  useEffect(() => {
+    const unsubscribe = onSnapshot(collection(db, localToken), (snapshot) => {
+      const snapshotDocs = [];
```

Comment:
Just a thought: 
Do we want to fetch all list here? could we just check the database for existing record and if it returns 'true' we set a notification if it returns 'false' we write to database. 
It's all connected to how we want to save users items to db, I'm just not sure that it is a good idea to run through the whole list, update the itemNames and then run through it again to check for matching items when we can run through the list only once or search inside the firebase directly. 
Looks like interesting topic to discuss

Reply:
We could either query the database for this one item, or use something like the following (from the Firebase docs):

```
// Assume we have the following data in the Database:
{
  "name": {
    "first": "Ada",
    "last": "Lovelace"
  }
}

// Test for the existence of certain keys within a DataSnapshot
var ref = firebase.database().ref("users/ada");
ref.once("value")
  .then(function(snapshot) {
    var a = snapshot.exists();  // true
    var b = snapshot.child("name").exists(); // true
    var c = snapshot.child("name/first").exists(); // true
    var d = snapshot.child("name/middle").exists(); // false
  });
```
This will return a boolean when we check it against the items in a collection...

Reply:
This is super interesting... Essentially this would replace the useEffect that is pulling the list data and also the find loop I believe? Would just need to normalize both inputs and I agree with the idea of having an imported utility function to do that